test/parser/LiteralOrIriTest.cpp

```diff
@@ -0,0 +1,184 @@
+  EXPECT_FALSE(literal.hasLanguageTag());
+  EXPECT_TRUE(literal.hasDatatype());
+  EXPECT_THAT("Hello World", asStringViewUnsafe(literal.getLiteralContent()));
+  EXPECT_THROW(literal.getLanguageTag(), ad_utility::Exception);
+  EXPECT_THAT("http://www.w3.org/2001/XMLSchema#string",
+              asStringViewUnsafe(literal.getDatatype().getContent()));
+}
+
+TEST(LiteralOrIri, LiteralOrIriWithLiteralWithQuotesAndDatatype) {
+  LiteralOrIri literal = LiteralOrIri::literalWithQuotesWithDatatype(
+      "\"Hello World\"",
+      Iri::iriref("<http://www.w3.org/2001/XMLSchema#string>"));
+
+  EXPECT_FALSE(literal.isIri());
+  EXPECT_THROW(literal.getIriContent(), ad_utility::Exception);
+  EXPECT_TRUE(literal.isLiteral());
+  EXPECT_FALSE(literal.hasLanguageTag());
+  EXPECT_TRUE(literal.hasDatatype());
+  EXPECT_THAT("Hello World", asStringViewUnsafe(literal.getLiteralContent()));
+  EXPECT_THROW(literal.getLanguageTag(), ad_utility::Exception);
+  EXPECT_THAT("http://www.w3.org/2001/XMLSchema#string",
+              asStringViewUnsafe(literal.getDatatype().getContent()));
+}
+
+TEST(LiteralOrIri, LiteralOrIriWithLiteralAndLanguageTag) {
+  LiteralOrIri literal =
+      LiteralOrIri::literalWithoutQuotesWithLanguageTag("Hej världen", "@se");
+
+  EXPECT_FALSE(literal.isIri());
+  EXPECT_THROW(literal.getIriContent(), ad_utility::Exception);
+  EXPECT_TRUE(literal.isLiteral());
+  EXPECT_TRUE(literal.hasLanguageTag());
+  EXPECT_FALSE(literal.hasDatatype());
+  EXPECT_THAT("Hej världen", asStringViewUnsafe(literal.getLiteralContent()));
+  EXPECT_THAT("se", asStringViewUnsafe(literal.getLanguageTag()));
+  EXPECT_THROW(literal.getDatatype(), ad_utility::Exception);
+}
+
+TEST(LiteralOrIri, LiteralOrIriWithLiteralWithQuotesAndLanguageTag) {
+  LiteralOrIri literal = LiteralOrIri::literalWithQuotesWithLanguageTag(
+      "'''Hej världen'''", "@se");
+
+  EXPECT_FALSE(literal.isIri());
+  EXPECT_THROW(literal.getIriContent(), ad_utility::Exception);
+  EXPECT_TRUE(literal.isLiteral());
+  EXPECT_TRUE(literal.hasLanguageTag());
+  EXPECT_FALSE(literal.hasDatatype());
+  EXPECT_THAT("Hej världen", asStringViewUnsafe(literal.getLiteralContent()));
+  EXPECT_THAT("se", asStringViewUnsafe(literal.getLanguageTag()));
+  EXPECT_THROW(literal.getDatatype(), ad_utility::Exception);
+}
+
+TEST(LiteralOrIri, GetContent) {
+  LiteralOrIri iri = LiteralOrIri::iriref("<https://example.org/books/book1>");
+  LiteralOrIri literalWithLanguageTag =
+      LiteralOrIri::literalWithoutQuotesWithLanguageTag("Hello World", "@de");
+  LiteralOrIri literalWithDatatype =
+      LiteralOrIri::literalWithoutQuotesWithDatatype(
+          "ABC", Iri::iriref("<https://example.org>"));
+
+  EXPECT_THAT("https://example.org/books/book1",
+              asStringViewUnsafe(iri.getContent()));
+  EXPECT_THAT("Hello World",
+              asStringViewUnsafe(literalWithLanguageTag.getContent()));
+  EXPECT_THAT("ABC", asStringViewUnsafe(literalWithDatatype.getContent()));
```

Comment:
Especially for the literals you should add tests, where the escaping actually does something, so where we could in theory detect if the wrong normalization was used.
(Don't exaggerate, the normalizations themselves are tested elsewhere, but just to make the normalizations visible).